### ISSUE TITLE
Impossible to delete "broken" event artifact notebook

### ISSUE THREAD

**Issue by misje** [Velocidex/velociraptor#3854]
In order to delete a event artifact notebook, you need to first navigate to the notebook tab in the events viewer, then click the delete button:

![2024-10-26_12-56](https://github.com/user-attachments/assets/a290604b-4da0-414d-bf11-04706d716c33)

However, if the notebook fails to load for some reason, and the whole page is blank, there is no way to delete the notebook. The only workaround is to delete the notebook files in the datastore. An example of how to break the notebook is to add a timestamp to the first column passed to "LineChart" (as opposed to an epoch value).

I would expect the notebook to be deleted when all of the corresponding event data is deleted using this button:
![2024-10-26_12-58](https://github.com/user-attachments/assets/9b98007c-cdd9-470a-b059-754a63b8de7b)

That is when *all* of it is deleted, not just a small time-limited amount. After all, it is very confusing to get an existing notebook loaded from completely new event data – along with the old StartTime and EndTime. In normal flows, deleting the flow also deletes the notebook (or at least makes it inaccessible, certainly not to be loaded into the next flow from the same artifact).

Having to delete all event data to delete a notebook isn't necessarily a solution for solving an issue with broken notebooks, but I assume those cases are rare. However, also deleting the notebook when all traces of the event artifact is gone, is far less surprising behaviour to the user.

**Comment by scudette** [comment]
Can you please check the JavaScript console for any exceptions? 

The interfaced should not be crashing

**Comment by misje** [comment]
I don't have the same dataset and cannot reproduce it any more, but I remember from the console output that there was a NaN exception of some sort. Definitely related to the plot, as it only crashed when it was attempted rendered.

**Comment by misje** [comment]
![2024-10-26_13-27](https://github.com/user-attachments/assets/d84bfa1b-674c-4822-bce1-7b19ef1f5256)

**Comment by scudette** [comment]
The javascript is minidied so it makes it hard to know where the error occurs. Can you please click on the link in the top right and the pretty print icon (looks like `{}` )  and attach the screenshot where the error occurs?

**Comment by misje** [comment]
Here is a very simple reproducible Markdown cell: `{{ Query "SELECT timestamp(epoch=now()) AS Ts FROM scope()" | LineChart }}`

**Comment by misje** [comment]
> Thanks for reporting it - it should be fixed by #3860

This solves the underlying issue that "broke" the interface. However, even when things don't break, I still feel that deleting the raw data in the event artifacts viewer should also delete the notebook. It's surprising behaviour to open the notebook tab for fresh data and not understand why StartTime/EndTime behaves as expected (I would have expected a new, fresh notebook when I delete everything), and why the default VQL is modified.

Can we not delete the notebook when pressing the "Delete" button? It matches the experience when deleting a flow/collection: I don't get an old notebook from my previous deleted flow when I enter the notebook tab for a collection.

**Comment by misje** [comment]
When I press the delete button, I always do it clear all data, being sure to zoom out sufficiently before pressing it. Is it not possible to determine whether the delete request actually deletes all available data, and if so, remove the notebook as well?

**Comment by scudette** [comment]
No I think this will be totally unexpected - deleting the data should be completely different from deleting the notebook. This is the way it is with other notebook as well which have different delete functions for data and notebook - although when the flow is deleted completely the notebook is also deleted incidentally since the notebook exists within the flow.

![image](https://github.com/user-attachments/assets/a23ada2d-84ea-40b4-b07d-70349d31f7d2)

**Comment by scudette** [comment]
Sorry I just checked - this is actually the way it is

![image](https://github.com/user-attachments/assets/467d1a6a-81a0-4ddf-9803-f85d2d3750c5)

**Comment by misje** [comment]
Yes, I am aware of the button. Perhaps my take on this right now is from artifact development, where I have had to delete data and the notebook heaps of times. From normal use, I imagine I rarely need to clear all data, and even when I do, I still want to keep my hard work in my notebook. So I agree, you're right, and my initial suggestion indeed has the most surprising behaviour.

**Comment by scudette** [comment]
I think the most confusing part of this is that notebook parameters like StartTime and EndTime are not obvious and can not be graphically changed so it is somewhat surprising when the notebook does not reflect the timeline's view. The user can adjust the timeline view but then it is hard to recalculate the notebook relative to that range (or even realize that the notebook is "static" and needs to recalculate to reflect the new range).

This is a wider issue which we should tackle in the next release in making notebooks more interactive by using the GUI to add parameters within the cells. We already have something similar with the "Format Columns" function - a GUI element which manipulates the VQL in the cell. We could extend the same idea to build out some of the VQL in a more structured way to be reflected in the GUI to help users adjust parameters.

**Comment by misje** [comment]
Indeed, that is one of the reasons that could result in a need for more notebook deletions than in earlier versions. I can easily imagine a confused user not understanding why the notebook "doesn't work", but deleting it and recreating it might fix the issue. My current remedy is to always include a `// LET StartTime = '[…]'` comment in artifact VQL noteboooks in order to make it easier for the user to understand that these variables can still be overridden like before.
